### ISSUE TITLE
Problems updating goa_<species> README files

### ISSUE THREAD

**Issue by tonysawfordebi** [geneontology/go-site#191]
I'm getting errors from SVN when I try to commit updated versions of the GOA species-specific README files:

$ svn commit -m "Updated to reflect new file names and content" goa_human.README 
svn: Commit failed (details follow):
svn: Entry '/nfs/gns/misc/misc03/nfs5-vol9-sp-pro2/go/goc-svn/gene-associations/readme/goa_human.README' has unexpectedly changed special status

Not sure what this means, or whether the problem is at the local or remote end

**Comment by jmcherry** [comment]
Tony,

In the old setup it seems you didn’t update those files.  It was added as a soft link in the local copy what was used for the website.  I’ve deleted the links listed below except for the two goa_uniprot files, their soft links remain in that local copy.  The goa_human.README and others were in the repository, not sure why they’d be in the repository and have softlinks.  I deleted from the repository the ones with soft links, except goa_uniprot.README and goa_uniprot_noiea.README.

I think you can svn add the files back and it will be okay.

Your updates to goa.README and goa_pdb.README happened okay.

-Mike

lrwxrwxrwx 1 gocvs gocvs    10 Nov  4  2014 goa_chicken.README -> goa.README
lrwxrwxrwx 1 gocvs gocvs    10 Nov  4  2014 goa_cow.README -> goa.README
lrwxrwxrwx 1 gocvs gocvs    10 Nov  4  2014 goa_dog.README -> goa.README
lrwxrwxrwx 1 gocvs gocvs    10 Oct  7  2014 goa_human.README -> goa.README
lrwxrwxrwx 1 gocvs gocvs    10 Nov  4  2014 goa_pig.README -> goa.README
lrwxrwxrwx 1 gocvs gocvs    10 Nov  4  2014 goa_ref_chicken.README -> goa.README
lrwxrwxrwx 1 gocvs gocvs    10 Nov  4  2014 goa_ref_cow.README -> goa.README
lrwxrwxrwx 1 gocvs gocvs    10 Nov  4  2014 goa_ref_dog.README -> goa.README
lrwxrwxrwx 1 gocvs gocvs    10 Oct  7  2014 goa_ref_human.README -> goa.README
lrwxrwxrwx 1 gocvs gocvs    10 Nov  4  2014 goa_ref_pig.README -> goa.README
lrwxrwxrwx 1 gocvs gocvs    10 Nov  4  2014 goa_uniprot_noiea.README -> goa.README
lrwxrwxrwx 1 gocvs gocvs    10 Nov  4  2014 goa_uniprot.README -> goa.README



On Jun 10, 2016, at 12:32 AM, Tony Sawford <notifications@github.com<mailto:notifications@github.com>> wrote:


I'm getting errors from SVN when I try to commit updated versions of the GOA species-specific README files:

$ svn commit -m "Updated to reflect new file names and content" goa_human.README
svn: Commit failed (details follow):
svn: Entry '/nfs/gns/misc/misc03/nfs5-vol9-sp-pro2/go/goc-svn/gene-associations/readme/goa_human.README' has unexpectedly changed special status

Not sure what this means, or whether the problem is at the local or remote end

**Comment by tonysawfordebi** [comment]
Hi Mike,

I just tried adding the files, with the following result:

svn: warning: 'goa_human.README' is already under version control
svn: warning: 'goa_chicken.README' is already under version control
svn: warning: 'goa_cow.README' is already under version control
svn: warning: 'goa_pig.README' is already under version control
svn: warning: 'goa_dog.README' is already under version control

The attempt to do a commit failed with the same error as I got previously.

**Comment by jmcherry** [comment]
Odd, I did this:

[root@cherry-vm57 readme]# touch goa_human.README
[root@cherry-vm57 readme]# sudo -u gocvs svn delete goa_human.README
svn: E200005: Use --force to override this restriction (local modifications may be lost)
svn: E200005: '/data/share/ftp/go/gene-associations/readme/goa_human.README' is not under version control

I guess we’ll need Stuart else to weigh in on this?

-Mike


On Jun 10, 2016, at 7:16 AM, Tony Sawford <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Mike,

I just tried adding the files, with the following result:

svn: warning: 'goa_human.README' is already under version control
svn: warning: 'goa_chicken.README' is already under version control
svn: warning: 'goa_cow.README' is already under version control
svn: warning: 'goa_pig.README' is already under version control
svn: warning: 'goa_dog.README' is already under version control

The attempt to do a commit failed with the same error as I got previously.

**Comment by jmcherry** [comment]
I should add.  I’ve just working from a local copy and I don’t think it has any special privileges.  You could be able to make this type of change too?


On Jun 10, 2016, at 7:22 AM, Mike Cherry <cherry@stanford.edu<mailto:cherry@stanford.edu>> wrote:

Odd, I did this:

[root@cherry-vm57 readme]# touch goa_human.README
[root@cherry-vm57 readme]# sudo -u gocvs svn delete goa_human.README
svn: E200005: Use --force to override this restriction (local modifications may be lost)
svn: E200005: '/data/share/ftp/go/gene-associations/readme/goa_human.README' is not under version control

I guess we’ll need Stuart else to weigh in on this?

-Mike


On Jun 10, 2016, at 7:16 AM, Tony Sawford <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Mike,

I just tried adding the files, with the following result:

svn: warning: 'goa_human.README' is already under version control
svn: warning: 'goa_chicken.README' is already under version control
svn: warning: 'goa_cow.README' is already under version control
svn: warning: 'goa_pig.README' is already under version control
svn: warning: 'goa_dog.README' is already under version control

The attempt to do a commit failed with the same error as I got previously.

**Comment by tonysawfordebi** [comment]
I'll try doing an svn remove and then a re-add to see whether it solves the problem; if not, then, as you say, I think it's time to invoke the Stuart option.

**Comment by tonysawfordebi** [comment]
Nope, not having much luck - I've tried deleting goa_pig.README, but SVN doesn't want to play ball:

$ svn remove --force goa_pig.README
D         goa_pig.README
$ svn commit -m "Deleted old README" goa_pig.README
svn: Commit failed (details follow):
svn: Aborting commit: '/nfs/gns/misc/misc03/nfs5-vol9-sp-pro2/go/goc-svn/gene-associations/readme/goa_pig.README' remains in conflict
$ svn status goa_pig.README
!     C goa_pig.README
      >   local delete, incoming delete upon update

**Comment by jmcherry** [comment]
Meatloaf is running 1.7.22 version of svn.  Stuart updated the local copy to days ago.  For the local copy on meatloaf I ran svn upgrade yesterday.  I don't think that should be involved because it was just the local copy.

From meatloaf and my laptop goa_human.README is not in the repository.

-Mike


Sent from my iPad

On Jun 10, 2016, at 7:39 AM, Tony Sawford <notifications@github.com<mailto:notifications@github.com>> wrote:


Nope, not having much luck - I've tried deleting goa_pig.README, but SVN doesn't want to play ball:

$ svn remove --force goa_pig.README
D goa_pig.README
$ svn commit -m "Deleted old README" goa_pig.README
svn: Commit failed (details follow):
svn: Aborting commit: '/nfs/gns/misc/misc03/nfs5-vol9-sp-pro2/go/goc-svn/gene-associations/readme/goa_pig.README' remains in conflict
$ svn status goa_pig.README
! C goa_pig.README
> local delete, incoming delete upon update

-

**Comment by kltm** [comment]
It might be worth noting that the SVN FS changed between versions during that period, with 1.6 being v10 and 1.7 being v12. I'm not sure if it's relevant in this particular case (we mostly ran into client issues around this for Jenkins), but it might be worth having kick around in the back of your heads.

**Comment by stuartmiyasato** [comment]
I'm hardly a svn expert, but I will look into this and get back to everyone.  Sorry Tony, your work day is pretty much over already, so I guess it won't be until next week before you see any changes/proposed solutions.

**Comment by stuartmiyasato** [comment]
@tonysawfordebi I'm not sure if this following post reflects your scenario, but it may be worth looking into:  

http://stackoverflow.com/questions/4317973/svn-how-to-resolve-local-edit-incoming-delete-upon-update-message

The explanation in the section "Long version:" seems like it applies in your case here, but without being able to see your local copy, I can't tell for sure.  See if those steps, particularly the "svn revert" command help fix anything.  If not, I'll dig deeper.

**Comment by kltm** [comment]
@jmcherry after locking all the svn commands onto compatible (same) version, "svn resolve" got things going again.

**Comment by tonysawfordebi** [comment]
I finally managed to get all the files updated, with a combination of svn revert, remove, and add commands - it was all pretty tedious, but we got there in the end.

Thanks for the help, all.

**Comment by jmcherry** [comment]
No one replied to my query from last week about the inclusion of the new goa files in the main directory.  As of today the filtering script is only acting on the goa_X.gaf.gz where X = human, dog, cow, chicken, pig

Committed to the submission directory are 3 other files for each of these 5 organisms:

goa_X_complex.gaf.gz
goa_X_isoform.gaf.gz
goa_X_rna.gaf.gz

I have not added the other 3 files as I am waiting to hear from someone that this will not impact the website or data loads.  I don’t think it will but I am waiting for comfirmation.

-Mike


On Jun 13, 2016, at 1:38 AM, Tony Sawford <notifications@github.com<mailto:notifications@github.com>> wrote:


I finally managed to get all the files updated, with a combination of svn revert, remove, and add commands - it was all pretty tedious, but we got there in the end.

Thanks for the help, all.
